app/services/billable_metrics/aggregations/unique_count_service.rb

```diff
@@ -39,6 +51,96 @@ def running_total(options)
 
         (1..result.aggregation).to_a
       end
+
+      private
+
+      attr_reader :from_datetime, :to_datetime
+
+      # This method fetches the latest event in current period. If such a event exists we know that metadata
+      # with previous aggregation and previous maximum aggregation are stored there. Fetching these metadata values
+      # would help us in pay in advance value calculation without iterating through all events in current period
+      def previous_event
+        @previous_event ||= begin
+          query = events_scope(from_datetime:, to_datetime:)
+            .joins(:quantified_event)
+            .where("#{sanitized_field_name} IS NOT NULL")
+            .where.not(id: event.id)
+            .where('quantified_events.added_at::timestamp(0) >= ?', from_datetime)
+            .where('quantified_events.added_at::timestamp(0) <= ?', to_datetime)
+            .order(created_at: :desc)
+
+          query
+            .where('quantified_events.removed_at::timestamp(0) IS NULL')
+            .or(
+              query
+                .where('quantified_events.removed_at::timestamp(0) >= ?', from_datetime)
+                .where('quantified_events.removed_at::timestamp(0) <= ?', to_datetime)
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)